lib/aws/rails/middleware/elastic_beanstalk_worker_listener.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+module Aws
+  module Rails
+    # Middleware to handle requests from the SQS Daemon present on Elastic Beanstalk worker environments.
+    class ElasticBeanstalkWorkerListener
+      INTERNAL_ERROR_MESSAGE = 'Failed to execute job - see Rails log for more details.'
+      INTERNAL_ERROR_RESPONSE = [500, { 'Content-Type' => 'text/plain' }, [INTERNAL_ERROR_MESSAGE]].freeze
+      FORBIDDEN_MESSAGE = 'Request with aws-sqsd user agent was made from untrusted address.'
+      FORBIDDEN_RESPONSE = [403, { 'Content-Type' => 'text/plain' }, [FORBIDDEN_MESSAGE]].freeze
+
+      def initialize(app)
+        @app = app
+        @logger = ActiveSupport::Logger.new(STDOUT)
+      end
+
+      def call(env)
+        request = ActionDispatch::Request.new env
+        if from_sqs_daemon?(request)
+          @logger.debug 'aws-rails-sdk middleware detected call from Elastic Beanstalk SQS Daemon.'
+
+          # Only accept requests from this user agent if it is from localhost or a docker host in case of forgery.
+          if request.local? || sent_from_docker_host?(request)
+            return periodic_task?(request) ? execute_periodic_task(request) : execute_job(request)
+          else
+            @logger.warn "SQSD request detected from untrusted address #{request.remote_ip}; returning 403 forbidden."
+            return FORBIDDEN_RESPONSE
+          end
+        end
+        @app.call(env)
+      end
+
+      private
+
+      def execute_job(request)
+        # Jobs queued from the Active Job SQS adapter contain the JSON message in the request body.
+        job = JSON.parse(request.body.string)
```

Comment:
I'd recommend using the [JobRunner](https://github.com/aws/aws-sdk-rails/blob/master/lib/aws/rails/sqs_active_job/job_runner.rb) here - it does functionally do the same thing, but lets us keep that logic abstracted.  It also uses the Aws::Json (which lets users use whichever json engine they have configured rather).

Reply:
Ah yeah makes sense - completely missed that class while working on this! Will update.

Reply:
Agree with your comments lower down on not using the JobRunner here since it takes the sqs message -  your update here looks good.